In [ ]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
df = pd.read_csv("/content/drive/MyDrive/Colab Notebooks/fake_job_postings.csv")

In [ ]:
df.head()

job_id                                      title            location  \
0       1                           Marketing Intern    US, NY, New York   
1       2  Customer Service - Cloud Video Production      NZ, , Auckland   
2       3    Commissioning Machinery Assistant (CMA)       US, IA, Wever   
3       4          Account Executive - Washington DC  US, DC, Washington   
4       5                        Bill Review Manager  US, FL, Fort Worth   

  department salary_range                                    company_profile  \
0  Marketing          NaN  We're Food52, and we've created a groundbreaki...   
1    Success          NaN  90 Seconds, the worlds Cloud Video Production ...   
2        NaN          NaN  Valor Services provides Workforce Solutions th...   
3      Sales          NaN  Our passion for improving quality of life thro...   
4        NaN          NaN  SpotSource Solutions LLC is a Global Human Cap...   

                                         description  \
0  Food52, a fast-growing, James Beard Award-winn...   
1  Organised - Focused - Vibrant - Awesome!Do you...   
2  Our client, located in Houston, is actively se...   
3  THE COMPANY: ESRI – Environmental Systems Rese...   
4  JOB TITLE: Itemization Review ManagerLOCATION:...   

                                        requirements  \
0  Experience with content management systems a m...   
1  What we expect from you:Your key responsibilit...   
2  Implement pre-commissioning and commissioning ...   
3  EDUCATION: Bachelor’s or Master’s in GIS, busi...   
4  QUALIFICATIONS:RN license in the State of Texa...   

                                            benefits  telecommuting  \
0                                                NaN              0   
1  What you will get from usThrough being part of...              0   
2                                                NaN              0   
3  Our culture is anything but corporate—we have ...              0   
4                              Full Benefits Offered              0   

   has_company_logo  has_questions employment_type required_experience  \
0                 1              0           Other          Internship   
1                 1              0       Full-time      Not Applicable   
2                 1              0             NaN                 NaN   
3                 1              0       Full-time    Mid-Senior level   
4                 1              1       Full-time    Mid-Senior level   

  required_education                   industry              function  \
0                NaN                        NaN             Marketing   
1                NaN  Marketing and Advertising      Customer Service   
2                NaN                        NaN                   NaN   
3  Bachelor's Degree          Computer Software                 Sales   
4  Bachelor's Degree     Hospital & Health Care  Health Care Provider   

   fraudulent  
0           0  
1           0  
2           0  
3           0  
4           0

In [ ]:
df.columns

Index(['job_id', 'title', 'location', 'department', 'salary_range',
       'company_profile', 'description', 'requirements', 'benefits',
       'telecommuting', 'has_company_logo', 'has_questions', 'employment_type',
       'required_experience', 'required_education', 'industry', 'function',
       'fraudulent'],
      dtype='object')

In [ ]:
df.shape

(17879, 18)

In [ ]:
df.isnull().sum()

job_id                     0
title                      0
location                 346
department             11546
salary_range           15011
company_profile         3307
description                0
requirements            2694
benefits                7209
telecommuting              0
has_company_logo           0
has_questions              0
employment_type         3470
required_experience     7049
required_education      8104
industry                4902
function                6454
fraudulent                 0
dtype: int64

In [ ]:
df['location'].value_counts()

GB, LND, London          718
US, NY, New York         658
US, CA, San Francisco    472
GR, I, Athens            464
US, ,                    339
                        ... 
GB, SFK, Leiston           1
GB, LND, Hammersmith       1
US, WA, Seattle            1
BE                         1
GB, WSX, Chichester        1
Name: location, Length: 3105, dtype: int64

# Removing Null values on Department

In [ ]:
df['department'].value_counts()

Sales                                          551
Engineering                                    487
Marketing                                      401
Operations                                     270
IT                                             225
                                              ... 
Commercial Management / Contract Management      1
Exec                                             1
Marcomm                                          1
CRM                                              1
Hospitality                                      1
Name: department, Length: 1337, dtype: int64

## Before

In [ ]:
df['department'].isnull().sum()

11546

In [ ]:
df['department']=df['department'].fillna('Any Department')

## After

In [ ]:
df['department'].isnull().sum()

0

# Removing Null values on Salary Range

In [ ]:
df['salary_range'].value_counts()

0-0            142
40000-50000     66
30000-40000     55
25000-30000     37
45000-67000     37
              ... 
15-25            1
60-75            1
27500-36000      1
20-22            1
3700-3800        1
Name: salary_range, Length: 874, dtype: int64

## Before

In [ ]:
df['salary_range'].isnull().sum()

15011

In [ ]:
salary_range_null = df[df['salary_range'].isnull()]

In [ ]:
salary_range_null.head()

job_id                                      title            location  \
0       1                           Marketing Intern    US, NY, New York   
1       2  Customer Service - Cloud Video Production      NZ, , Auckland   
2       3    Commissioning Machinery Assistant (CMA)       US, IA, Wever   
3       4          Account Executive - Washington DC  US, DC, Washington   
4       5                        Bill Review Manager  US, FL, Fort Worth   

       department salary_range  \
0       Marketing          NaN   
1         Success          NaN   
2  Any Department          NaN   
3           Sales          NaN   
4  Any Department          NaN   

                                     company_profile  \
0  We're Food52, and we've created a groundbreaki...   
1  90 Seconds, the worlds Cloud Video Production ...   
2  Valor Services provides Workforce Solutions th...   
3  Our passion for improving quality of life thro...   
4  SpotSource Solutions LLC is a Global Human Cap...   

                                         description  \
0  Food52, a fast-growing, James Beard Award-winn...   
1  Organised - Focused - Vibrant - Awesome!Do you...   
2  Our client, located in Houston, is actively se...   
3  THE COMPANY: ESRI – Environmental Systems Rese...   
4  JOB TITLE: Itemization Review ManagerLOCATION:...   

                                        requirements  \
0  Experience with content management systems a m...   
1  What we expect from you:Your key responsibilit...   
2  Implement pre-commissioning and commissioning ...   
3  EDUCATION: Bachelor’s or Master’s in GIS, busi...   
4  QUALIFICATIONS:RN license in the State of Texa...   

                                            benefits  telecommuting  \
0                                                NaN              0   
1  What you will get from usThrough being part of...              0   
2                                                NaN              0   
3  Our culture is anything but corporate—we have ...              0   
4                              Full Benefits Offered              0   

   has_company_logo  has_questions employment_type required_experience  \
0                 1              0           Other          Internship   
1                 1              0       Full-time      Not Applicable   
2                 1              0             NaN                 NaN   
3                 1              0       Full-time    Mid-Senior level   
4                 1              1       Full-time    Mid-Senior level   

  required_education                   industry              function  \
0                NaN                        NaN             Marketing   
1                NaN  Marketing and Advertising      Customer Service   
2                NaN                        NaN                   NaN   
3  Bachelor's Degree          Computer Software                 Sales   
4  Bachelor's Degree     Hospital & Health Care  Health Care Provider   

   fraudulent  
0           0  
1           0  
2           0  
3           0  
4           0

## Salary Range Based on Fraudulent

In [ ]:
salary_range_null_yes = salary_range_null[salary_range_null['fraudulent']==1]

In [ ]:
salary_range_null_yes.shape

(642, 18)

In [ ]:
salary_range_null_no = salary_range_null[salary_range_null['fraudulent']==0]

In [ ]:
salary_range_null_no.shape

(14369, 18)

In [ ]:
df['salary_range'] = df['salary_range'].fillna("Based on skill")

In [ ]:
df['salary_range'].value_counts()

Based on skill    15011
0-0                 142
40000-50000          66
30000-40000          55
45000-67000          37
                  ...  
15-25                 1
60-75                 1
300000-325000         1
20-22                 1
3700-3800             1
Name: salary_range, Length: 875, dtype: int64

## After

In [ ]:
df['salary_range'].isnull().sum()

0

# Removing Null values on description

In [ ]:
df['description'].isnull().sum()

0

In [ ]:
df.shape

(17879, 18)

In [ ]:
df[df['description'].isnull()]

Empty DataFrame
Columns: [job_id, title, location, department, salary_range, company_profile, description, requirements, benefits, telecommuting, has_company_logo, has_questions, employment_type, required_experience, required_education, industry, function, fraudulent]
Index: []

In [ ]:
df=df.drop(index=17513)

In [ ]:
df.shape

(17878, 18)

# Removing Null values on Employement Type

In [ ]:
df['employment_type'].value_counts()

Full-time    11620
Contract      1524
Part-time      797
Temporary      241
Other          227
Name: employment_type, dtype: int64

In [ ]:
df['employment_type'].isnull().sum()

3469

In [ ]:
df[df['employment_type'].isnull()]

job_id                                              title  \
2           3            Commissioning Machinery Assistant (CMA)   
5           6                                   Accounting Clerk   
7           8                   Lead Guest Service Specialist      
11         12      Talent Sourcer (6 months fixed-term contract)   
17         18  Southend-on-Sea Traineeships Under NAS 16-18 Y...   
...       ...                                                ...   
17847   17848                                 SEO CONTENT WRITER   
17852   17853                      Call Center/Customer Service    
17854   17855                                   Android Engineer   
17858   17859     Immediate Requirement: iOS Developer for Qatar   
17863   17864                      Next Generation Depth Sensing   

                       location        department    salary_range  \
2                 US, IA, Wever    Any Department  Based on skill   
5                      US, MD,     Any Department  Based on skill   
7         US, CA, San Francisco    Any Department  Based on skill   
11              GB, LND, London                HR  Based on skill   
17     GB, SOS, Southend-on-Sea    Any Department  Based on skill   
...                         ...               ...             ...   
17847           GB, LND, london               SEO  Based on skill   
17852            US, NJ, Newark  Customer Service  Based on skill   
17854     US, CA, San Francisco       Engineering  Based on skill   
17858                  QA, DA,     Any Department  Based on skill   
17863         US, MA, Cambridge                MM  Based on skill   

                                         company_profile  \
2      Valor Services provides Workforce Solutions th...   
5                                                    NaN   
7      Airenvy’s mission is to provide lucrative yet ...   
11     Want to build a 21st century financial service...   
17     Established on the principles that full time e...   
...                                                  ...   
17847                                                NaN   
17852  At Command we care enough to consistently plac...   
17854  Shyp is the easiest way to send anything, anyw...   
17858                                                NaN   
17863  MERL's internship program gives students excel...   

                                             description  \
2      Our client, located in Houston, is actively se...   
5      Job OverviewApex is an environmental consultin...   
7      Who is Airenvy?Hey there! We are seasoned entr...   
11     TransferWise is the clever new way to move mon...   
17     Government funding is only available for 16-18...   
...                                                  ...   
17847  A Blogger or Journalist is required for delive...   
17852  At Command we care enough to consistently plac...   
17854  What we're looking for:You've been an integral...   
17858  Greetings from VAM SYSTEMS…..VAM SYSTEMS is a ...   
17863  MERL is looking for a well qualified individua...   

                                            requirements  \
2      Implement pre-commissioning and commissioning ...   
5                                                    NaN   
7      Experience with CRM software, live chat, and p...   
11     We’re looking for someone who:Proven track rec...   
17     16-18 year olds only due to government funding...   
...                                                  ...   
17847  Key responsibilities within this role:Supporti...   
17852  Responsibilities:Determines requirements by wo...   
17854  Track record of developing native Android appl...   
17858   iOS programming “Objective – C “, C++#URL_01a...   
17863                                                NaN   

                                                benefits  telecommuting  \
2                                                    NaN              0   
5                                               

In [ ]:
df['employment_type'] = df['employment_type'].fillna('Not Mentioned')

## After

In [ ]:
df['employment_type'].isnull().sum()

0

# Removing null values on Required Experience

In [ ]:
df['required_experience'].value_counts()

Mid-Senior level    3809
Entry level         2697
Associate           2297
Not Applicable      1116
Director             389
Internship           381
Executive            141
Name: required_experience, dtype: int64

## Before

In [ ]:
df['required_experience'].isnull().sum()

7048

In [ ]:
df['required_experience'] = df['required_experience'].fillna("Not Required")

## After

In [ ]:
df['required_experience'].isnull().sum()

0

# Removing Null values on Required Education

In [ ]:
df['required_education'].value_counts()

Bachelor's Degree                    5145
High School or equivalent            2080
Unspecified                          1397
Master's Degree                       416
Associate Degree                      274
Certification                         170
Some College Coursework Completed     102
Professional                           74
Vocational                             49
Some High School Coursework            27
Doctorate                              26
Vocational - HS Diploma                 9
Vocational - Degree                     6
Name: required_education, dtype: int64

## Before

In [ ]:
df['required_education'].isnull().sum()

8103

In [ ]:
req_edu=df[df['required_education'].isnull()]

In [ ]:
req_edu.head()

job_id                                      title               location  \
0       1                           Marketing Intern       US, NY, New York   
1       2  Customer Service - Cloud Video Production         NZ, , Auckland   
2       3    Commissioning Machinery Assistant (CMA)          US, IA, Wever   
5       6                           Accounting Clerk               US, MD,    
7       8           Lead Guest Service Specialist     US, CA, San Francisco   

       department    salary_range  \
0       Marketing  Based on skill   
1         Success  Based on skill   
2  Any Department  Based on skill   
5  Any Department  Based on skill   
7  Any Department  Based on skill   

                                     company_profile  \
0  We're Food52, and we've created a groundbreaki...   
1  90 Seconds, the worlds Cloud Video Production ...   
2  Valor Services provides Workforce Solutions th...   
5                                                NaN   
7  Airenvy’s mission is to provide lucrative yet ...   

                                         description  \
0  Food52, a fast-growing, James Beard Award-winn...   
1  Organised - Focused - Vibrant - Awesome!Do you...   
2  Our client, located in Houston, is actively se...   
5  Job OverviewApex is an environmental consultin...   
7  Who is Airenvy?Hey there! We are seasoned entr...   

                                        requirements  \
0  Experience with content management systems a m...   
1  What we expect from you:Your key responsibilit...   
2  Implement pre-commissioning and commissioning ...   
5                                                NaN   
7  Experience with CRM software, live chat, and p...   

                                            benefits  telecommuting  \
0                                                NaN              0   
1  What you will get from usThrough being part of...              0   
2                                                NaN              0   
5                                                NaN              0   
7  Competitive Pay. You'll be able to eat steak e...              0   

   has_company_logo  has_questions employment_type required_experience  \
0                 1              0           Other          Internship   
1                 1              0       Full-time      Not Applicable   
2                 1              0   Not Mentioned        Not Required   
5                 0              0   Not Mentioned        Not Required   
7                 1              1   Not Mentioned        Not Required   

  required_education                   industry          function  fraudulent  
0                NaN                        NaN         Marketing           0  
1                NaN  Marketing and Advertising  Customer Service           0  
2                NaN                        NaN               NaN           0  
5                NaN                        NaN               NaN           0  
7                NaN                        NaN               NaN           0

## Based on Fraudulent

### Total null values on Required Education is 8104

In [ ]:
req_edu_yes=req_edu[req_edu['fraudulent']==1]

In [ ]:
req_edu_yes.shape

(450, 18)

In [ ]:
req_edu_no=req_edu[req_edu['fraudulent']==0]

In [ ]:
req_edu_no.shape

(7653, 18)

In [ ]:
df['required_education'] = df['required_education'].fillna("Not Mentioned")

## After

In [ ]:
df['required_education'].isnull().sum()

0

In [ ]:
df.isnull().sum()

job_id                    0
title                     0
location                346
department                0
salary_range              0
company_profile        3306
description               0
requirements           2694
benefits               7208
telecommuting             0
has_company_logo          0
has_questions             0
employment_type           0
required_experience       0
required_education        0
industry               4901
function               6453
fraudulent                0
dtype: int64

# Removing Null values on Industry

In [ ]:
df['industry'].value_counts()

Information Technology and Services    1734
Computer Software                      1376
Internet                               1062
Marketing and Advertising               828
Education Management                    822
                                       ... 
Shipbuilding                              1
Sporting Goods                            1
Museums and Institutions                  1
Wine and Spirits                          1
Ranching                                  1
Name: industry, Length: 131, dtype: int64

## Before

In [ ]:
df['industry'].isnull().sum()

4901

In [ ]:
df['industry'] = df['industry'].fillna('Not Mentioned')

## After

In [ ]:
df['industry'].isnull().sum()

0

In [ ]:
df['industry'].value_counts()

Not Mentioned                          4901
Information Technology and Services    1734
Computer Software                      1376
Internet                               1062
Marketing and Advertising               828
                                       ... 
Shipbuilding                              1
Sporting Goods                            1
Museums and Institutions                  1
Wine and Spirits                          1
Ranching                                  1
Name: industry, Length: 132, dtype: int64

# Removing null values on Function

## Before

In [ ]:
df['function'].isnull().sum()

6453

In [ ]:
df['function'].value_counts()

Information Technology    1749
Sales                     1468
Engineering               1348
Customer Service          1229
Marketing                  830
Administrative             630
Design                     340
Health Care Provider       338
Other                      325
Education                  325
Management                 317
Business Development       228
Accounting/Auditing        212
Human Resources            205
Project Management         183
Finance                    172
Consulting                 144
Writing/Editing            132
Art/Creative               132
Production                 116
Product Management         114
Quality Assurance          111
Advertising                 90
Business Analyst            84
Data Analyst                82
Public Relations            76
Manufacturing               74
General Business            68
Research                    50
Legal                       47
Strategy/Planning           46
Training                    38
Supply C

In [ ]:
df['function'] = df['function'].fillna("Any other")

## After

In [ ]:
df['function'].isnull().sum()

0

In [ ]:
df['function'].value_counts()

Any other                 6453
Information Technology    1749
Sales                     1468
Engineering               1348
Customer Service          1229
Marketing                  830
Administrative             630
Design                     340
Health Care Provider       338
Other                      325
Education                  325
Management                 317
Business Development       228
Accounting/Auditing        212
Human Resources            205
Project Management         183
Finance                    172
Consulting                 144
Writing/Editing            132
Art/Creative               132
Production                 116
Product Management         114
Quality Assurance          111
Advertising                 90
Business Analyst            84
Data Analyst                82
Public Relations            76
Manufacturing               74
General Business            68
Research                    50
Legal                       47
Strategy/Planning           46
Training

# Removing Null values on Location


## Before

In [ ]:
df['location'].isnull().sum()

346

In [ ]:
df['location'].value_counts()

GB, LND, London          718
US, NY, New York         658
US, CA, San Francisco    472
GR, I, Athens            464
US, ,                    339
                        ... 
GB, SFK, Leiston           1
GB, LND, Hammersmith       1
US, WA, Seattle            1
BE                         1
GB, WSX, Chichester        1
Name: location, Length: 3105, dtype: int64

## After

In [ ]:
df['location'] = df['location'].fillna('Unknown')

In [ ]:
df['location'].value_counts().sum()

17878

In [ ]:
df['location'].isnull().sum()

0

# Removing Null values on Company Profile

In [ ]:
df['company_profile'].value_counts().sum()

14572

## Before

In [ ]:
df['company_profile'].isnull().sum()

3306

In [ ]:
df['company_profile'] = df['company_profile'].fillna('No Profile')

## After

In [ ]:
df['company_profile'].isnull().sum()

0

# Removing Null values on Requirements

In [ ]:
df['requirements'].value_counts().sum()

15184

## Before

In [ ]:
df['requirements'].isnull().sum()

2694

In [ ]:
df['requirements'] = df['requirements'].fillna('No requirements needed')

## After

In [ ]:
df['requirements'].isnull().sum()

0

# Removing Null values on Benefits

In [ ]:
df['benefits'].value_counts()

See job description                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         726
Career prospects.                                                                                                                                                                                                                                                                                       

## Before

In [ ]:
df['benefits'].isnull().sum()

7208

In [ ]:
df['benefits'] = df['benefits'].fillna('Not Mentioned')

## After

In [ ]:
df['benefits'].isnull().sum()

0

In [ ]:
df.isnull().sum()

job_id                 0
title                  0
location               0
department             0
salary_range           0
company_profile        0
description            0
requirements           0
benefits               0
telecommuting          0
has_company_logo       0
has_questions          0
employment_type        0
required_experience    0
required_education     0
industry               0
function               0
fraudulent             0
dtype: int64

In [ ]:
df.head()

job_id                                      title            location  \
0       1                           Marketing Intern    US, NY, New York   
1       2  Customer Service - Cloud Video Production      NZ, , Auckland   
2       3    Commissioning Machinery Assistant (CMA)       US, IA, Wever   
3       4          Account Executive - Washington DC  US, DC, Washington   
4       5                        Bill Review Manager  US, FL, Fort Worth   

       department    salary_range  \
0       Marketing  Based on skill   
1         Success  Based on skill   
2  Any Department  Based on skill   
3           Sales  Based on skill   
4  Any Department  Based on skill   

                                     company_profile  \
0  We're Food52, and we've created a groundbreaki...   
1  90 Seconds, the worlds Cloud Video Production ...   
2  Valor Services provides Workforce Solutions th...   
3  Our passion for improving quality of life thro...   
4  SpotSource Solutions LLC is a Global Human Cap...   

                                         description  \
0  Food52, a fast-growing, James Beard Award-winn...   
1  Organised - Focused - Vibrant - Awesome!Do you...   
2  Our client, located in Houston, is actively se...   
3  THE COMPANY: ESRI – Environmental Systems Rese...   
4  JOB TITLE: Itemization Review ManagerLOCATION:...   

                                        requirements  \
0  Experience with content management systems a m...   
1  What we expect from you:Your key responsibilit...   
2  Implement pre-commissioning and commissioning ...   
3  EDUCATION: Bachelor’s or Master’s in GIS, busi...   
4  QUALIFICATIONS:RN license in the State of Texa...   

                                            benefits  telecommuting  \
0                                      Not Mentioned              0   
1  What you will get from usThrough being part of...              0   
2                                      Not Mentioned              0   
3  Our culture is anything but corporate—we have ...              0   
4                              Full Benefits Offered              0   

   has_company_logo  has_questions employment_type required_experience  \
0                 1              0           Other          Internship   
1                 1              0       Full-time      Not Applicable   
2                 1              0   Not Mentioned        Not Required   
3                 1              0       Full-time    Mid-Senior level   
4                 1              1       Full-time    Mid-Senior level   

  required_education                   industry              function  \
0      Not Mentioned              Not Mentioned             Marketing   
1      Not Mentioned  Marketing and Advertising      Customer Service   
2      Not Mentioned              Not Mentioned             Any other   
3  Bachelor's Degree          Computer Software                 Sales   
4  Bachelor's Degree     Hospital & Health Care  Health Care Provider   

   fraudulent  
0           0  
1           0  
2           0  
3           0  
4           0

In [ ]:
df.to_csv('data1.csv')

In [ ]:
df['fraudulent'].value_counts()

0    17013
1      865
Name: fraudulent, dtype: int64

In [ ]:
!pip install nltk

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/


In [ ]:
import nltk

In [ ]:
nltk.download('all')

[nltk_data] Downloading collection 'all'
[nltk_data]    | 
[nltk_data]    | Downloading package abc to /root/nltk_data...
[nltk_data]    |   Unzipping corpora/abc.zip.
[nltk_data]    | Downloading package alpino to /root/nltk_data...
[nltk_data]    |   Unzipping corpora/alpino.zip.
[nltk_data]    | Downloading package averaged_perceptron_tagger to
[nltk_data]    |     /root/nltk_data...
[nltk_data]    |   Unzipping taggers/averaged_perceptron_tagger.zip.
[nltk_data]    | Downloading package averaged_perceptron_tagger_ru to
[nltk_data]    |     /root/nltk_data...
[nltk_data]    |   Unzipping
[nltk_data]    |       taggers/averaged_perceptron_tagger_ru.zip.
[nltk_data]    | Downloading package basque_grammars to
[nltk_data]    |     /root/nltk_data...
[nltk_data]    |   Unzipping grammars/basque_grammars.zip.
[nltk_data]    | Downloading package bcp47 to /root/nltk_data...
[nltk_data]    | Downloading package biocreative_ppi to
[nltk_data]    |     /root/nltk_data...
[nltk_data]    |   U

True

In [ ]:
import re

In [ ]:
from nltk.stem import WordNetLemmatizer
le = WordNetLemmatizer()

In [ ]:
from nltk.corpus import stopwords

In [ ]:
data = pd.read_csv("/content/data1.csv")
datacp = pd.read_csv("/content/data1.csv")

In [ ]:
data.columns

Index(['Unnamed: 0', 'job_id', 'title', 'location', 'department',
       'salary_range', 'company_profile', 'description', 'requirements',
       'benefits', 'telecommuting', 'has_company_logo', 'has_questions',
       'employment_type', 'required_experience', 'required_education',
       'industry', 'function', 'fraudulent'],
      dtype='object')

In [ ]:
datacp.columns

Index(['Unnamed: 0', 'job_id', 'title', 'location', 'department',
       'salary_range', 'company_profile', 'description', 'requirements',
       'benefits', 'telecommuting', 'has_company_logo', 'has_questions',
       'employment_type', 'required_experience', 'required_education',
       'industry', 'function', 'fraudulent'],
      dtype='object')

In [ ]:
Description =[]

for i in range(0,len(data)):
  description = re.sub('[^a-zA-Z]'," ",data["description"][i])
  description = description.lower()
  description = description.split()


  description = [le.lemmatize(word) for word in description if not word in stopwords.words('english')]
  description = " ".join(description)
  Description.append(description)



In [ ]:
Description

['food fast growing james beard award winning online food community crowd sourced curated recipe hub currently interviewing full part time unpaid intern work small team editor executive developer new york city headquarters reproducing repackaging existing food content number partner site huffington post yahoo buzzfeed various content management systemsresearching blog website provision food affiliate programassisting day day affiliate program support screening affiliate assisting affiliate inquiriessupporting pr amp event neededhelping office administrative work filing mailing preparing meetingsworking developer document bug suggest improvement sitesupporting marketing executive staff',
 'organised focused vibrant awesome passion customer service slick typing skill maybe account management think administration cooler polar bear jetski need hear cloud video production service opperating glodal level yeah pretty cool serious delivering world class product excellent customer service rapid

In [ ]:
from sklearn.feature_extraction.text import TfidfVectorizer

In [ ]:
tv = TfidfVectorizer(max_features=2500)

In [ ]:
description1 = tv.fit_transform(Description).toarray()

In [ ]:
description1

array([[0., 0., 0., ..., 0., 0., 0.],
       [0., 0., 0., ..., 0., 0., 0.],
       [0., 0., 0., ..., 0., 0., 0.],
       ...,
       [0., 0., 0., ..., 0., 0., 0.],
       [0., 0., 0., ..., 0., 0., 0.],
       [0., 0., 0., ..., 0., 0., 0.]])

In [ ]:
datacp['description'] = description1

In [ ]:
datacp['description']

0        0.0
1        0.0
2        0.0
3        0.0
4        0.0
        ... 
17873    0.0
17874    0.0
17875    0.0
17876    0.0
17877    0.0
Name: description, Length: 17878, dtype: float64

In [ ]:
datacp.head()

Unnamed: 0  job_id                                      title  \
0           0       1                           Marketing Intern   
1           1       2  Customer Service - Cloud Video Production   
2           2       3    Commissioning Machinery Assistant (CMA)   
3           3       4          Account Executive - Washington DC   
4           4       5                        Bill Review Manager   

             location      department    salary_range  \
0    US, NY, New York       Marketing  Based on skill   
1      NZ, , Auckland         Success  Based on skill   
2       US, IA, Wever  Any Department  Based on skill   
3  US, DC, Washington           Sales  Based on skill   
4  US, FL, Fort Worth  Any Department  Based on skill   

                                     company_profile  description  \
0  We're Food52, and we've created a groundbreaki...          0.0   
1  90 Seconds, the worlds Cloud Video Production ...          0.0   
2  Valor Services provides Workforce Solutions th...          0.0   
3  Our passion for improving quality of life thro...          0.0   
4  SpotSource Solutions LLC is a Global Human Cap...          0.0   

                                        requirements  \
0  Experience with content management systems a m...   
1  What we expect from you:Your key responsibilit...   
2  Implement pre-commissioning and commissioning ...   
3  EDUCATION: Bachelor’s or Master’s in GIS, busi...   
4  QUALIFICATIONS:RN license in the State of Texa...   

                                            benefits  telecommuting  \
0                                      Not Mentioned              0   
1  What you will get from usThrough being part of...              0   
2                                      Not Mentioned              0   
3  Our culture is anything but corporate—we have ...              0   
4                              Full Benefits Offered              0   

   has_company_logo  has_questions employment_type required_experience  \
0                 1              0           Other          Internship   
1                 1              0       Full-time      Not Applicable   
2                 1              0   Not Mentioned        Not Required   
3                 1              0       Full-time    Mid-Senior level   
4                 1              1       Full-time    Mid-Senior level   

  required_education                   industry              function  \
0      Not Mentioned              Not Mentioned             Marketing   
1      Not Mentioned  Marketing and Advertising      Customer Service   
2      Not Mentioned              Not Mentioned             Any other   
3  Bachelor's Degree          Computer Software                 Sales   
4  Bachelor's Degree     Hospital & Health Care  Health Care Provider   

   fraudulent  
0           0  
1           0  
2           0  
3           0  
4           0

In [ ]:
Companyprofile =[]

for i in range(0,len(data)):
  cmpr = re.sub('[^a-zA-Z]'," ",data["company_profile"][i])
  cmpr = cmpr.lower()
  cmpr = cmpr.split()


  cmpr = [le.lemmatize(word) for word in cmpr if not word in stopwords.words('english')]
  cmpr = " ".join(cmpr)
  Companyprofile.append(cmpr)

In [ ]:
Companyprofile

['food created groundbreaking award winning cooking site support connect celebrate home cook give everything need one place top editorial business engineering team focused using technology find new better way connect people around specific food interest offer superb highly curated information food cooking attract talented home cook contributor country also publish well known professional like mario batali gwyneth paltrow danny meyer partnership whole food market random house food named best food website james beard foundation iacp featured new york time npr pando daily techcrunch today show located chelsea new york city',
 'second world cloud video production service second world cloud video production service enabling brand agency get high quality online video content shot produced anywhere world second make video production fast affordable managed seamlessly cloud purchase publish http url fbe afac cd c f b eef e e f ca dd second remove hassle cost risk speed issue working regular vi

In [ ]:
Companyprofile1 = tv.fit_transform(Companyprofile).toarray()

In [ ]:
Companyprofile1

array([[0.        , 0.        , 0.        , ..., 0.        , 0.        ,
        0.        ],
       [0.        , 0.05589091, 0.        , ..., 0.        , 0.        ,
        0.        ],
       [0.        , 0.        , 0.        , ..., 0.        , 0.        ,
        0.        ],
       ...,
       [0.        , 0.        , 0.        , ..., 0.        , 0.        ,
        0.        ],
       [0.        , 0.        , 0.        , ..., 0.        , 0.        ,
        0.        ],
       [0.        , 0.        , 0.        , ..., 0.        , 0.        ,
        0.        ]])

In [ ]:
datacp['company_profile'] = Companyprofile1

In [ ]:
datacp.head()

Unnamed: 0  job_id                                      title  \
0           0       1                           Marketing Intern   
1           1       2  Customer Service - Cloud Video Production   
2           2       3    Commissioning Machinery Assistant (CMA)   
3           3       4          Account Executive - Washington DC   
4           4       5                        Bill Review Manager   

             location      department    salary_range  company_profile  \
0    US, NY, New York       Marketing  Based on skill              0.0   
1      NZ, , Auckland         Success  Based on skill              0.0   
2       US, IA, Wever  Any Department  Based on skill              0.0   
3  US, DC, Washington           Sales  Based on skill              0.0   
4  US, FL, Fort Worth  Any Department  Based on skill              0.0   

   description                                       requirements  \
0          0.0  Experience with content management systems a m...   
1          0.0  What we expect from you:Your key responsibilit...   
2          0.0  Implement pre-commissioning and commissioning ...   
3          0.0  EDUCATION: Bachelor’s or Master’s in GIS, busi...   
4          0.0  QUALIFICATIONS:RN license in the State of Texa...   

                                            benefits  telecommuting  \
0                                      Not Mentioned              0   
1  What you will get from usThrough being part of...              0   
2                                      Not Mentioned              0   
3  Our culture is anything but corporate—we have ...              0   
4                              Full Benefits Offered              0   

   has_company_logo  has_questions employment_type required_experience  \
0                 1              0           Other          Internship   
1                 1              0       Full-time      Not Applicable   
2                 1              0   Not Mentioned        Not Required   
3                 1              0       Full-time    Mid-Senior level   
4                 1              1       Full-time    Mid-Senior level   

  required_education                   industry              function  \
0      Not Mentioned              Not Mentioned             Marketing   
1      Not Mentioned  Marketing and Advertising      Customer Service   
2      Not Mentioned              Not Mentioned             Any other   
3  Bachelor's Degree          Computer Software                 Sales   
4  Bachelor's Degree     Hospital & Health Care  Health Care Provider   

   fraudulent  
0           0  
1           0  
2           0  
3           0  
4           0

In [ ]:
Requirements =[]

for i in range(0,len(data)):
  req = re.sub('[^a-zA-Z]'," ",data["requirements"][i])
  req = req.lower()
  req = req.split()


  req = [le.lemmatize(word) for word in req if not word in stopwords.words('english')]
  req = " ".join(req)
  Requirements.append(req)

In [ ]:
Requirements

['experience content management system major plus blogging count familiar food editorial voice aestheticloves food appreciates importance home cooking cooking seasonsmeticulous editor perfectionist obsessive attention detail maddened typo broken link delighted finding fixing themcheerful pressureexcellent communication skillsa multi tasker juggler responsibility big smallinterested engaged social medium like twitter facebook pinterestloves problem solving collaborating drive food forwardthinks big picture pitch nitty gritty running small company dish shopping administrative support comfortable reality working startup call evening weekend working long hour',
 'expect key responsibility communicate client second team freelance community throughout video production process including shoot planning securing freelance talent managing workflow online production management system aim manage video project effectively produce great video client love key attributesclient focused excellent custom

In [ ]:
Requirements1 = tv.fit_transform(Requirements).toarray()

In [ ]:
Requirements1

array([[0.        , 0.        , 0.        , ..., 0.        , 0.        ,
        0.        ],
       [0.        , 0.        , 0.04200711, ..., 0.        , 0.        ,
        0.        ],
       [0.        , 0.        , 0.        , ..., 0.        , 0.        ,
        0.        ],
       ...,
       [0.        , 0.        , 0.04455527, ..., 0.        , 0.        ,
        0.        ],
       [0.        , 0.        , 0.        , ..., 0.        , 0.        ,
        0.        ],
       [0.        , 0.        , 0.        , ..., 0.        , 0.        ,
        0.        ]])

In [ ]:
datacp['requirements']  = Requirements1

In [ ]:
Benefits =[]

for i in range(0,len(data)):
  ben = re.sub('[^a-zA-Z]'," ",data["benefits"][i])
  ben = req.lower()
  ben = req.split()


  ben = [le.lemmatize(word) for word in ben if not word in stopwords.words('english')]
  ben = " ".join(ben)
  Benefits.append(ben)

In [ ]:
Benefits

['want hear depth understanding oo programmingyou totally get mvc multi tier architectureyou experience tdd writing automated testsyou understand relational database know write sqlyou worked agile style team cooperationbonus point worked retail beforeif sound like get touch delight u unique application huge project pipeline year need best talent board help achieve goal vend totally open receiving application people based overseas recruitment process technical talent lengthy involves lot face face time candidate existing team member start initial stage interviewing using awesome technology like skype google hangout really need hung u auckland making job offer',
 'want hear depth understanding oo programmingyou totally get mvc multi tier architectureyou experience tdd writing automated testsyou understand relational database know write sqlyou worked agile style team cooperationbonus point worked retail beforeif sound like get touch delight u unique application huge project pipeline year 

In [ ]:
Benefits1 = tv.fit_transform(Benefits).toarray()

In [ ]:
Benefits1

array([[0.09622504, 0.09622504, 0.19245009, ..., 0.09622504, 0.09622504,
        0.09622504],
       [0.09622504, 0.09622504, 0.19245009, ..., 0.09622504, 0.09622504,
        0.09622504],
       [0.09622504, 0.09622504, 0.19245009, ..., 0.09622504, 0.09622504,
        0.09622504],
       ...,
       [0.09622504, 0.09622504, 0.19245009, ..., 0.09622504, 0.09622504,
        0.09622504],
       [0.09622504, 0.09622504, 0.19245009, ..., 0.09622504, 0.09622504,
        0.09622504],
       [0.09622504, 0.09622504, 0.19245009, ..., 0.09622504, 0.09622504,
        0.09622504]])

In [ ]:
datacp['benefits'] = Benefits1

In [ ]:
Department =[]

for i in range(0,len(data)):
  dep = re.sub('[^a-zA-Z]'," ",data["department"][i])
  dep = dep.lower()
  dep = dep.split()


  dep = [le.lemmatize(word) for word in dep if not word in stopwords.words('english')]
  dep = " ".join(dep)
  Department.append(dep)

In [ ]:
Department

['marketing',
 'success',
 'department',
 'sale',
 'department',
 'department',
 'androidpit',
 'department',
 'department',
 'department',
 'department',
 'hr',
 'department',
 'department',
 'sale',
 'sale',
 'r',
 'department',
 'department',
 'department',
 'department',
 'department',
 'engagement',
 'businessfriend com',
 'department',
 'department',
 'marketing',
 'medical',
 'department',
 'department',
 'department',
 'department',
 'field',
 'department',
 '',
 'department',
 'department',
 'department',
 'design',
 'department',
 'department',
 'department',
 'department',
 'department',
 'department',
 'department',
 'department',
 'department',
 'department',
 'department',
 'production',
 'department',
 'department',
 'department',
 'department',
 'department',
 'department',
 'icm',
 'department',
 'department',
 'department',
 'general service',
 'department',
 'department',
 'department',
 'department',
 'engineering',
 'department',
 'department',
 '',
 'marketing',
 

In [ ]:
Department1 = tv.fit_transform(Department).toarray()

In [ ]:
Department1

array([[0., 0., 0., ..., 0., 0., 0.],
       [0., 0., 0., ..., 0., 0., 0.],
       [0., 0., 0., ..., 0., 0., 0.],
       ...,
       [0., 0., 0., ..., 0., 0., 0.],
       [0., 0., 0., ..., 0., 0., 0.],
       [0., 0., 0., ..., 0., 0., 0.]])

In [ ]:
datacp['department'] = Department1

In [ ]:
Industry =[]

for i in range(0,len(data)):
  ind = re.sub('[^a-zA-Z]'," ",data["industry"][i])
  ind = ind.lower()
  ind = ind.split()


  ind = [le.lemmatize(word) for word in ind if not word in stopwords.words('english')]
  ind = " ".join(ind)
  Industry.append(ind)

In [ ]:
Industry1 = tv.fit_transform(Industry).toarray()

In [ ]:
Industry1

array([[0.        , 0.        , 0.        , ..., 0.        , 0.        ,
        0.        ],
       [0.        , 0.        , 0.70710678, ..., 0.        , 0.        ,
        0.        ],
       [0.        , 0.        , 0.        , ..., 0.        , 0.        ,
        0.        ],
       ...,
       [0.        , 0.        , 0.        , ..., 0.        , 0.        ,
        0.        ],
       [0.        , 0.        , 0.        , ..., 0.        , 0.        ,
        0.        ],
       [0.        , 0.        , 0.        , ..., 0.        , 0.        ,
        0.        ]])

In [ ]:
datacp['industry'] = Industry1

In [ ]:
datacp.head()

Unnamed: 0  job_id                                      title  \
0           0       1                           Marketing Intern   
1           1       2  Customer Service - Cloud Video Production   
2           2       3    Commissioning Machinery Assistant (CMA)   
3           3       4          Account Executive - Washington DC   
4           4       5                        Bill Review Manager   

             location  department    salary_range  company_profile  \
0    US, NY, New York         0.0  Based on skill              0.0   
1      NZ, , Auckland         0.0  Based on skill              0.0   
2       US, IA, Wever         0.0  Based on skill              0.0   
3  US, DC, Washington         0.0  Based on skill              0.0   
4  US, FL, Fort Worth         0.0  Based on skill              0.0   

   description  requirements  benefits  telecommuting  has_company_logo  \
0          0.0           0.0  0.096225              0                 1   
1          0.0           0.0  0.096225              0                 1   
2          0.0           0.0  0.096225              0                 1   
3          0.0           0.0  0.096225              0                 1   
4          0.0           0.0  0.096225              0                 1   

   has_questions employment_type required_experience required_education  \
0              0           Other          Internship      Not Mentioned   
1              0       Full-time      Not Applicable      Not Mentioned   
2              0   Not Mentioned        Not Required      Not Mentioned   
3              0       Full-time    Mid-Senior level  Bachelor's Degree   
4              1       Full-time    Mid-Senior level  Bachelor's Degree   

   industry              function  fraudulent  
0       0.0             Marketing           0  
1       0.0      Customer Service           0  
2       0.0             Any other           0  
3       0.0                 Sales           0  
4       0.0  Health Care Provider           0

In [ ]:
from sklearn.preprocessing import LabelEncoder
encoder = LabelEncoder()

In [ ]:
datacp['employment_type'] = encoder.fit_transform(datacp['employment_type'])
mapping=dict(zip(encoder.classes_,encoder.transform(encoder.classes_)))
print(datacp['employment_type'],':',mapping)

0        3
1        1
2        2
3        1
4        1
        ..
17873    1
17874    1
17875    1
17876    0
17877    1
Name: employment_type, Length: 17878, dtype: int64 : {'Contract': 0, 'Full-time': 1, 'Not Mentioned': 2, 'Other': 3, 'Part-time': 4, 'Temporary': 5}


In [ ]:
datacp['required_experience'] = encoder.fit_transform(datacp['required_experience'])
mapping=dict(zip(encoder.classes_,encoder.transform(encoder.classes_)))
print(datacp['required_experience'],':',mapping)

0        4
1        6
2        7
3        5
4        5
        ..
17873    5
17874    5
17875    7
17876    6
17877    5
Name: required_experience, Length: 17878, dtype: int64 : {'Associate': 0, 'Director': 1, 'Entry level': 2, 'Executive': 3, 'Internship': 4, 'Mid-Senior level': 5, 'Not Applicable': 6, 'Not Required': 7}


In [ ]:
datacp['required_education'] = encoder.fit_transform(datacp['required_education'])
mapping=dict(zip(encoder.classes_,encoder.transform(encoder.classes_)))
print(datacp['required_education'],':',mapping)

0        6
1        6
2        6
3        1
4        1
        ..
17873    6
17874    1
17875    6
17876    7
17877    6
Name: required_education, Length: 17878, dtype: int64 : {'Associate Degree': 0, "Bachelor's Degree": 1, 'Certification': 2, 'Doctorate': 3, 'High School or equivalent': 4, "Master's Degree": 5, 'Not Mentioned': 6, 'Professional': 7, 'Some College Coursework Completed': 8, 'Some High School Coursework': 9, 'Unspecified': 10, 'Vocational': 11, 'Vocational - Degree': 12, 'Vocational - HS Diploma': 13}


In [ ]:
datacp.loc[datacp['salary_range']=="Dec-25",'salary_range']='Based on skill'

In [ ]:
datacp.loc[datacp['salary_range']=="Jun-18",'salary_range']='Based on skill'

In [ ]:
datacp.loc[datacp['salary_range']=="Oct-15",'salary_range']='Based on skill'

In [ ]:
datacp.loc[datacp['salary_range']=="Oct-20",'salary_range']='Based on skill'

In [ ]:
datacp.loc[datacp['salary_range']=="02-Apr",'salary_range']='Based on skill'

In [ ]:
datacp.loc[datacp['salary_range']=="03-Apr",'salary_range']='Based on skill'

In [ ]:
datacp.loc[datacp['salary_range']=="04-Apr",'salary_range']='Based on skill'

In [ ]:
datacp.loc[datacp['salary_range']=="04-Jun",'salary_range']='Based on skill'

In [ ]:
datacp.loc[datacp['salary_range']=="08-Sep",'salary_range']='Based on skill'

In [ ]:
datacp.loc[datacp['salary_range']=="09-Dec",'salary_range']='Based on skill'

In [ ]:
datacp.loc[datacp['salary_range']=="10-Nov",'salary_range']='Based on skill'

In [ ]:
datacp.loc[datacp['salary_range']=="10-Oct",'salary_range']='Based on skill'

In [ ]:
datacp['salary_range'] = encoder.fit_transform(datacp['salary_range'])
mapping=dict(zip(encoder.classes_,encoder.transform(encoder.classes_)))
print(datacp['salary_range'],':',mapping)

0        862
1        862
2        862
3        862
4        862
        ... 
17873    862
17874    862
17875    862
17876    862
17877    862
Name: salary_range, Length: 17878, dtype: int64 : {'0-0': 0, '0-1': 1, '0-1000': 2, '0-100000': 3, '0-110406': 4, '0-115': 5, '0-120000': 6, '0-1200000': 7, '0-12500': 8, '0-130000': 9, '0-15000': 10, '0-150000': 11, '0-16000': 12, '0-180000': 13, '0-2000': 14, '0-20000': 15, '0-24000': 16, '0-25000': 17, '0-268': 18, '0-30000': 19, '0-32000': 20, '0-34300': 21, '0-35000': 22, '0-38000': 23, '0-43500': 24, '0-45000': 25, '0-48000': 26, '0-50000': 27, '0-60000': 28, '0-65000': 29, '0-70000': 30, '0-90000': 31, '0-92000': 32, '0-9360000': 33, '0-95000': 34, '02-Jun': 35, '100-120': 36, '100-150': 37, '100-200': 38, '1000-1000': 39, '1000-10000': 40, '1000-1100': 41, '1000-1400': 42, '1000-1500': 43, '1000-2000': 44, '1000-20000': 45, '1000-26000': 46, '1000-3000': 47, '1000-4000': 48, '1000-5000': 49, '1000-6000': 50, '10000-10000': 51, '10000-100

In [ ]:
datacp['function'] = encoder.fit_transform(datacp['function'])
mapping=dict(zip(encoder.classes_,encoder.transform(encoder.classes_)))
print(datacp['function'],':',mapping)

0        23
1         8
2         3
3        32
4        17
         ..
17873    32
17874     0
17875     3
17876    10
17877    13
Name: function, Length: 17878, dtype: int64 : {'Accounting/Auditing': 0, 'Administrative': 1, 'Advertising': 2, 'Any other': 3, 'Art/Creative': 4, 'Business Analyst': 5, 'Business Development': 6, 'Consulting': 7, 'Customer Service': 8, 'Data Analyst': 9, 'Design': 10, 'Distribution': 11, 'Education': 12, 'Engineering': 13, 'Finance': 14, 'Financial Analyst': 15, 'General Business': 16, 'Health Care Provider': 17, 'Human Resources': 18, 'Information Technology': 19, 'Legal': 20, 'Management': 21, 'Manufacturing': 22, 'Marketing': 23, 'Other': 24, 'Product Management': 25, 'Production': 26, 'Project Management': 27, 'Public Relations': 28, 'Purchasing': 29, 'Quality Assurance': 30, 'Research': 31, 'Sales': 32, 'Science': 33, 'Strategy/Planning': 34, 'Supply Chain': 35, 'Training': 36, 'Writing/Editing': 37}


In [ ]:
fea = datacp.drop(['Unnamed: 0','job_id','title','location','benefits','has_company_logo','has_questions','telecommuting'],axis=1)

In [ ]:
fea.shape

(17878, 11)

In [ ]:
fea.head()

department  salary_range  company_profile  description  requirements  \
0         0.0           862              0.0          0.0           0.0   
1         0.0           862              0.0          0.0           0.0   
2         0.0           862              0.0          0.0           0.0   
3         0.0           862              0.0          0.0           0.0   
4         0.0           862              0.0          0.0           0.0   

   employment_type  required_experience  required_education  industry  \
0                3                    4                   6       0.0   
1                1                    6                   6       0.0   
2                2                    7                   6       0.0   
3                1                    5                   1       0.0   
4                1                    5                   1       0.0   

   function  fraudulent  
0        23           0  
1         8           0  
2         3           0  
3        32           0  
4        17           0

In [ ]:
x = fea.drop(['fraudulent'],axis=1)
y = fea.fraudulent

from sklearn.model_selection import train_test_split
x_train,x_test,y_train,y_test = train_test_split(x,y,train_size = 0.7,random_state = 0,stratify=y)

In [ ]:
x_train.shape

(12514, 10)

In [ ]:
from sklearn.feature_selection import VarianceThreshold
var = VarianceThreshold(threshold=0)
var.fit(x_train)

VarianceThreshold(threshold=0)

In [ ]:
len(x_train.columns[var.get_support()])

10

In [ ]:
x_train.head()

department  salary_range  company_profile  description  requirements  \
15535         0.0           862              0.0     0.069201           0.0   
16098         0.0           407              0.0     0.000000           0.0   
436           0.0           862              0.0     0.000000           0.0   
5640          0.0           862              0.0     0.000000           0.0   
12816         0.0           862              0.0     0.000000           0.0   

       employment_type  required_experience  required_education  industry  \
15535                1                    7                   6       0.0   
16098                1                    0                  10       0.0   
436                  0                    7                   1       0.0   
5640                 1                    7                   6       0.0   
12816                2                    7                   6       0.0   

       function  
15535         3  
16098         4  
436           3  
5640         32  
12816        24

In [ ]:
from sklearn.svm import SVC

svc = SVC()
svc.fit(x_train,y_train)

SVC()

In [ ]:
svc_ans =svc.predict(x_test)

In [ ]:
from sklearn.metrics import accuracy_score
SVC_score = accuracy_score(y_test,svc_ans)*100
print(SVC_score)

95.15287099179717


In [ ]:
from sklearn.ensemble import RandomForestClassifier

Rand = RandomForestClassifier(n_estimators=100)
Rand.fit(x_train,y_train)

RandomForestClassifier()

In [ ]:
Rand_pred = Rand.predict(x_test)

In [ ]:
Rand_score = accuracy_score(y_test,Rand_pred)*100
print(Rand_score)

95.917225950783


In [ ]:
DEPARTMENT = input("Enter the Department:")
COMPANY_PROFILE = input("Enter the Company Profile:")
DESCRIPTION = input("Enter the Description :")
REQUIREMENTS = input("Enter the  Requirements:")
INDUSTRY = input("Enter the Industry:")
SALARY_RANGE = input("Enter the salary_range :")
EMPLOYEE_TYPE =input("Enter the Employee_Type {'Contract': 0, 'Full-time': 1, 'Not Mentioned': 2, 'Other': 3, 'Part-time': 4, 'Temporary': 5} :")
REQUIRED_EXPERIENCE = input("Enter the required_Experience  {'Associate': 0, 'Director': 1, 'Entry level': 2, 'Executive': 3, 'Internship': 4, 'Mid-Senior level': 5, 'Not Applicable': 6, 'Not Required': 7} :")
REQUIRED_EDUCATION = input("Enter the Required_Education  {'Associate Degree': 0, 'Bachelor s Degree': 1, 'Certification': 2, 'Doctorate': 3, 'High School or equivalent': 4, 'Masters Degree': 5, 'Not Mentioned': 6, 'Professional': 7, 'Some College Coursework Completed': 8, 'Some High School Coursework': 9, 'Unspecified': 10, 'Vocational': 11, 'Vocational - Degree': 12, 'Vocational - HS Diploma': 13} :")
FUNCTION = input("Enter the Function :")

Enter the Department:Marketing
Enter the Company Profile:We're Food52, and we've created a groundbreaking and award-winning cooking site. We support, connect, and celebrate home cooks, and give them everything they need in one place.We have a top editorial, business, and engineering team. We're focused on using technology to find new and better ways to connect people around their specific food interests, and to offer them superb, highly curated information about food and cooking. We attract the most talented home cooks and contributors in the country; we also publish well-known professionals like Mario Batali, Gwyneth Paltrow, and Danny Meyer. And we have partnerships with Whole Foods Market and Random House.Food52 has been named the best food website by the James Beard Foundation and IACP, and has been featured in the New York Times, NPR, Pando Daily, TechCrunch, and on the Today Show.We're located in Chelsea, in New York City.
Enter the Description :Food52, a fast-growing, James Bear

In [ ]:
import re
import numpy as np

DEP = []
DEPARTMENT = 'Marketing'
#for i in range(len(DEPARTMENT)):
dep1 = re.sub('[^a-zA-Z]', ' ', DEPARTMENT)
dep1 = dep1.lower()
dep1 = dep1.split()
dep1 = [le.lemmatize(word) for word in dep1 if word not in stopwords.words('english')]
dep1 = ' '.join(dep1)
DEP.append(dep1)

In [ ]:
DEPARTMENT = tv.fit_transform(DEP).toarray()

In [ ]:
DEPARTMENT

array([[1.]])

In [ ]:
CMP =[]

#for i in range(0,len(COMPANY_PROFILE)):
cmpr1 = re.sub('[^a-zA-Z]'," ",COMPANY_PROFILE)
cmpr1 = cmpr1.lower()
cmpr1 = cmpr1.split()


cmpr1 = [le.lemmatize(word) for word in cmpr1 if not word in stopwords.words('english')]
cmpr1 = " ".join(cmpr1)
CMP.append(cmpr1)

In [ ]:
COMPANY_PROFILE = tv.fit_transform(CMP).toarray()

In [ ]:
COMPANY_PROFILE = np.reshape(COMPANY_PROFILE,(78,1))

In [ ]:
COMPANY_PROFILE.shape

(78,)

In [ ]:
COMPANY_PROFILE = np.concatenate(COMPANY_PROFILE)

In [ ]:
COMPANY_PROFILE = np.mean(COMPANY_PROFILE)

In [ ]:
COMPANY_PROFILE

0.09894149142837044

In [ ]:
DES =[]

#for i in range(0,len(DESCRIPTION)):
des1 = re.sub('[^a-zA-Z]'," ",DESCRIPTION)
des1 = des1.lower()
des1 = des1.split()


des1 = [le.lemmatize(word) for word in des1 if not word in stopwords.words('english')]
des1 = " ".join(des1)
DES.append(des1)



In [ ]:
DESCRIPTION = tv.fit_transform(DES).toarray()

In [ ]:
DESCRIPTION = DESCRIPTION.reshape(73,1)

In [ ]:
DESCRIPTION = np.concatenate(DESCRIPTION)

In [ ]:
DESCRIPTION = np.mean(DESCRIPTION)

In [ ]:
DESCRIPTION

0.10592913411432761

In [ ]:
REQ =[]

#for i in range(0,len(REQUIREMENTS)):
req1 = re.sub('[^a-zA-Z]'," ",REQUIREMENTS)
req1 = req1.lower()
req1 = req1.split()


req1 = [le.lemmatize(word) for word in req1 if not word in stopwords.words('english')]
req1 = " ".join(req1)
REQ.append(req1)

In [ ]:
REQUIREMENTS = tv.fit_transform(REQ).toarray()

In [ ]:
REQUIREMENTS = REQUIREMENTS.reshape(72,1)

In [ ]:
REQUIREMENTS.shape

(72, 1)

In [ ]:
#REQUIREMENTS = np.concatenate(REQUIREMENTS)
REQUIREMENTS = np.mean(REQUIREMENTS)
REQUIREMENTS

0.11336088438956907

In [ ]:
IND =[]

#for i in range(0,len(INDUSTRY)):
ind1 = re.sub('[^a-zA-Z]'," ",INDUSTRY)
ind1 = ind1.lower()
ind1 = ind1.split()


ind1 = [le.lemmatize(word) for word in ind1 if not word in stopwords.words('english')]
ind1 = " ".join(ind1)
IND.append(ind1)

In [ ]:
INDUSTRY = tv.fit_transform(IND).toarray()

In [ ]:
INDUSTRY

array([[1.]])

	salary_range	company_profile	description	requirements	employment_type	required_experience	required_education	industry	function

In [ ]:
FINAL_DATA =[[DEPARTMENT,SALARY_RANGE,COMPANY_PROFILE,DESCRIPTION,REQUIREMENTS,EMPLOYEE_TYPE,REQUIRED_EXPERIENCE,REQUIRED_EDUCATION,INDUSTRY,FUNCTION]]
FINAL_DATA

[[array([[1.]]),
  '7',
  0.09894149142837044,
  0.10592913411432761,
  0.11336088438956907,
  '3',
  '2',
  '4',
  array([[1.]]),
  '5']]

In [ ]:
FRAUDULENT = Rand.predict(FINAL_DATA)

/usr/local/lib/python3.10/dist-packages/sklearn/base.py:439: UserWarning: X does not have valid feature names, but RandomForestClassifier was fitted with feature names
  warnings.warn(


In [ ]:
print(FRAUDULENT)

[1]
